feat(module:modal): global settings for the modal via the forRoot method

volvachev opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

Issue Number: #7458

## What is the new behavior?

The `NzModalModule` now has a `forRoot` method, where you can set default settings for all kinds of `modal`.

The settings object can be changed as it is not clear which settings can be global and which can't.


## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->

simplejason commented:
I think you can use `NZ_CONFIG` instead of setting these properties with `forRoot()`

volvachev commented:
> I think you can use `NZ_CONFIG` instead of setting these properties with `forRoot()`

Yes, you're right. I have updated pr, added a new field to the nz config.

simplejason commented:
> > I think you can use `NZ_CONFIG` instead of setting these properties with `forRoot()`
> 
> Yes, you're right. I have updated pr, added a new field to the nz config.

Please make sure all tests work as expected

simplejason commented:
@hsuanxyz Please take time to review :)

tegola commented:
@volvachev are you able to add more tests for lines that have been marked as not covered?
https://github.com/NG-ZORRO/ng-zorro-antd/pull/7461/checks?check_run_id=6652702744
